visivo/parsers/mkdocs_utils/markdown.py

```diff
@@ -0,0 +1,154 @@
+from textwrap import dedent
+import yaml
+
+
+def find_refs(obj):
+    refs = []
+    if isinstance(obj, dict):
+        for key, value in obj.items():
+            if key == "$ref":
+                refs.append(value)
+            elif isinstance(value, (dict, list)):
+                refs.extend(find_refs(value))
+    elif isinstance(obj, list):
+        for item in obj:
+            if isinstance(item, (dict, list)):
+                refs.extend(find_refs(item))
+    return sorted(refs)
+
+
+def handle_attribute_properties(model_defs: dict, attribute_property_object: dict):
+    """Handle extracting type, description, default from different types of property objects that pydantic produces"""
+    type, description, default = (
+        "",
+        attribute_property_object.get("description", ""),
+        "None",
+    )
+
+    attribute_key_type = ".".join(list(attribute_property_object.keys()))
+    if "allOf" in attribute_key_type:
```

Comment:
Might be an opportunity to break this method down further.  Each of the if statements could be 

```
if "allOf":
  type, description, default = extract_all_of_properties(attribute_key_type)
. . .
```
 This would make it easier to test.

Reply:
That's a good idea I'll break these out.